lib/ansible/modules/network/meraki/meraki_firewalled_services.py

```diff
@@ -0,0 +1,241 @@
+  meraki_firewalled_services:
+    auth_key: abc123
+    state: present
+    org_name: YourOrg
+    net_name: YourNet
+    service: web
+    access: restricted
+    allowed_ips:
+      - 192.0.1.1
+      - 192.0.1.2
+  delegate_to: localhost
+
+- name: Query appliance services
+  meraki_firewalled_services:
+    auth_key: abc123
+    state: query
+    org_name: YourOrg
+    net_name: YourNet
+  delegate_to: localhost
+
+- name: Query services
+  meraki_firewalled_services:
+    auth_key: abc123
+    state: query
+    org_name: YourOrg
+    net_name: YourNet
+    service: ICMP
+  delegate_to: localhost
+'''
+
+RETURN = r'''
+data:
+    description: List of network services.
+    returned: info
+    type: complex
+    contains:
+      access:
+        description: Access assigned to a service type.
+        returned: success
+        type: str
+        sample: unrestricted
+      service:
+        description: Service to apply policy to.
+        returned: success
+        type: str
+        sample: ICMP
+      allowed_ips:
+        description: List of IP addresses to have access to service.
+        returned: success
+        type: str
+        sample: 192.0.1.0
+'''
+
+import os
+from ansible.module_utils.basic import AnsibleModule, json, env_fallback
+from ansible.module_utils.urls import fetch_url
+from ansible.module_utils._text import to_native
+from ansible.module_utils.common.dict_transformations import recursive_diff
+from ansible.module_utils.network.meraki.meraki import MerakiModule, meraki_argument_spec
+
+
+def main():
+
+    # define the available arguments/parameters that a user can pass to
+    # the module
+
+    argument_spec = meraki_argument_spec()
+    argument_spec.update(
+        net_id=dict(type='str'),
+        net_name=dict(type='str', aliases=['network']),
+        state=dict(type='str', default='present', choices=['query', 'present']),
+        service=dict(type='str', default=None, choices=['ICMP', 'SNMP', 'web']),
+        access=dict(type='str', choices=['blocked', 'restricted', 'unrestricted']),
+        allowed_ips=dict(type='list', element='str'),
+    )
+
+    # the AnsibleModule object will be our abstraction working with Ansible
+    # this includes instantiation, a couple of common attr would be the
+    # args/params passed to the execution, as well as if the module
+    # supports check mode
+    module = AnsibleModule(argument_spec=argument_spec,
+                           supports_check_mode=True,
+                           )
+
+    meraki = MerakiModule(module, function='firewalled_services')
+    module.params['follow_redirects'] = 'all'
+
+    net_services_urls = {'firewalled_services': '/networks/{net_id}/firewalledServices'}
+    services_urls = {'firewalled_services': '/networks/{net_id}/firewalledServices/{service}'}
+
+    meraki.url_catalog['network_services'] = net_services_urls
+    meraki.url_catalog['service'] = services_urls
+
+    if meraki.params['net_name'] and meraki.params['net_id']:
```

Comment:
Can this be added while creating an object of AnsibleModule?
Something similar to 
```
mutually_exclusive = [('net_name',  'net_id')]
    module = AnsibleModule(argument_spec=argument_spec,
                           supports_check_mode=True,
                           mutually_exclusive=mutually_exclusive
                           )

```

Reply:
Fixed. Thank you.